website/content/docs/rfcs/self-init.mdx

```diff
@@ -0,0 +1,292 @@
+token, no access to OpenBao is possible as no authentication methods are
+configured.
+
+Thus, simply initializing OpenBao and removing the need for the root token's
+one-off use is insufficient unless recovery shares are placed elsewhere (in
+a secure location) and used on subsequent configuration steps. Changes to
+recovery key handling will be discussed in a subsequent RFC.
+
+Ideally, initialization should set up an audit method, an initial policy, and
+an auth method which would be used by the administrator via any declarative
+policy. Then subsequent configuration could be entirely declarative and not
+feature any one-time steps or one-time authentication methods. Only when an
+operator is ready should one-time steps be taken, such as generating and
+preserving long-term access to recovery keys.
+
+However, each auth method and organization's policy requirements are
+different. While we cannot seek to fully replace a declarative framework like
+OpenTofu, we should have sufficient flexibility to define all initialization
+up-front. Adding to this, output or request chaining is necessary: if OpenBao
+creates a certificate authority, we may wish to make that available on disk
+for other automation to use and deploy or reuse it in subsequent requests to
+the instance during initialization (to e.g., configure a certificate auth
+mount).
+
+To bound the security risk, initialization configuration must only be applied
+on first startup. It should not be treated as the long-term owner of these
+resources and an import to a proper IaC tool would be preferable long-term.
+
+We need this to interact nicely with existing configuration management
+techniques; HCL and JSON-equivalent representations should be possible for
+any initialization and we should not attempt to build a new DSL for this.
+
+Lastly, whatever format is chosen should ultimately form the basis of the
+[profile system](/blog/profiles/) and should ideally have some code reuse
+with it.
+
+## User-facing Description
+
+Users will define one or more initialization blocks:
+
+```hcl
+initialize "identity" {
+  request "mount-userpass" {
+    operation = "update"
+    path = "sys/auth/userpass"
+    data = {
+      type = "kv"
+      path = "userpass/"
+      description = "admin
+    }
+  }
+
+  request "userpass-add-admin" {
+    operation = "update"
+    path = "auth/userpass/users/admin"
+    data = {
+      "password" = {
+        type = "string"
+        source = "env"
+        env_var = "INITIAL_ADMIN_PASSWORD"
+      }
+      "token_policies" = ["superuser"]
+    }
+  }
+}
+
+initialize "policy" {
+  request "add-superuser-policy" {
+    operation = "update"
+    path = "sys/policies/acl/superuser"
+    data = {
+      policy = << EOP
+path "*" {
+  capabilities = ["create", "update", "read", "delete", "list", "scan", "sudo"]
+}
+EOP
+    }
+  }
+}
+```
+
+Equivalently in JSON this looks like:
+
+```json
```

Comment:
@KrzysztofKornalewski-Reply This is an equivalent JSON configuration if you want to test it. :-)